frontend/src/components/grantaccess/GrantAccess.vue

```diff
@@ -209,224 +209,201 @@ function roleSelected(evt: any) {
-                                        :value="domainOptions.BCEID"
-                                        @change="userDomainChange()"
-                                        class="p-radiobutton"
-                                    />
-                                    <label
-                                        class="mx-2 form-check-label"
-                                        for="becidSelect"
-                                        >BCeID</label
-                                    >
-                                </div>
-                            </div>
+                        <label> Select user's domain </label>
+                        <div class="px-0">
+                            <RadioButton
+                                v-model="formData.domain"
+                                :checked="isIdirDomainSelected()"
+                                inputId="idirSelect"
+                                name="domainRadioOptions"
+                                :value="domainOptions.IDIR"
+                                @change="userDomainChange()"
+                            />
+                            <label class="mx-2" for="idirSelect">IDIR</label>
+                        </div>
+                        <div class="px-0">
+                            <RadioButton
+                                v-model="formData.domain"
+                                inputId="becidSelect"
+                                name="domainRadioOptions"
+                                :value="domainOptions.BCEID"
+                                @change="userDomainChange()"
+                            />
+                            <label class="mx-2" for="becidSelect">BCeID</label>
                         </div>
                     </div>
 
-                    <div class="form-group">
-                        <div class="row mb-0">
-                            <label for="userIdInput" class="control-label px-0"
-                                >Type user's domain name
-                                <span class="text-danger"> *</span></label
+                    <div class="row">
+                        <label for="userIdInput"
+                            >Type user's id
+                            <span class="text-danger"> *</span></label
+                        >
+                        <div class="mt-0 col-md-3 px-0">
+                            <Field
+                                name="userId"
+                                :validateOnChange="true"
+                                v-model="formData.userId"
+                                v-slot="{ field }"
                             >
-                        </div>
-
-                        <div class="row mt-0">
-                            <div class="col-md-3 px-0">
-                                <Field
+                                <InputText
                                     id="userIdInput"
```

Comment:
For all the InputText field, we can add an id to it, and let the label for attributes use the InputText id.